esileapclient/v1/resource_fields.py

```diff
@@ -0,0 +1,70 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+class Resource(object):
+    """Resource class
+    This class is used to manage the various fields that a resource (e.g.
+    Offer) contains.  An individual field consists of a
+    'field_id' (key) and a 'label' (value).  The caller only provides the
+    'field_ids' when instantiating the object.
+    Ordering of the 'field_ids' will be preserved as specified by the caller.
+    It also provides the ability to exclude some of these fields when they are
+    being used for sorting.
+    """
+
+    FIELDS = {
+        'created_at': "Created At",
+        'end_date': "End Date",
+        'id': "ID",
+        'project_id': "Project ID",
+        'properties': "Properties",
+        'resource_type': "Resource Type",
+        'resource_uuid': "Resource UUID",
+        'start_date': "Start Date",
+        'status': "Status",
+        'updated_at': "Updated At",
+        'uuid': "UUID",
+    }
+
+    def __init__(self, field_ids):
+        """Create a Resource object
+        :param field_ids:  A list of strings that the Resource object will
+                           contain.  Each string must match an existing key in
+                           FIELDS.
+        :raises: ValueError if sort_excluded or override_labels contains values
+                 not in field_ids
+        """
+
+        self._fields = tuple(field_ids)
+        self._labels = tuple([self.FIELDS[x] for x in field_ids])
+
+    @property
+    def fields(self):
+        return self._fields
+
+    @property
+    def labels(self):
+        return self._labels
+
+
+OFFER_RESOURCE = Resource(
```

Comment:
It looks like the order of these fields controls the output order. If so, is there a way to control the order for the list operation? For example, in ironic a 'node show' displays attributes in alphabetical order, but 'node list' controls the order of fields:

+--------------------------------------+---------+---------------+-------------+--------------------+-------------+
| UUID                                 | Name    | Instance UUID | Power State | Provisioning State | Maintenance |
+--------------------------------------+---------+---------------+-------------+--------------------+-------------+
| 9fd5bc1a-fc95-44ff-a586-2878c7bc7bf8 | dell-14 | None          | power off   | active             | False       |
| 32341d5e-f841-4381-bcfb-b7e54751baaf | dell-15 | None          | None        | enroll             | False       |
+--------------------------------------+---------+---------------+-------------+--------------------+-------------+

I also think that we *never* show the 'id'. That's a strictly internal field.

Reply:
esi-leap returns it. So I'm assuming thats an issue that also would have to be fixed on the esi-leap end of things

Reply:
esi-leap returns it, but it's not something the client needs to display. I think removing 'id' from OFFER_RESOURCE will remove it.

Reply:
I can add in this sorting if needed for this pr

Reply:
Yeah, I think that'd be a good idea!